### PR TITLE
Update shuffle algorithm

### PR DESCRIPTION
**Pull request by rjzamora**
This PR makes the changes to `_shuffle_df` and `_add_data_scatter` that were discussed/motivated in [this comment](https://github.com/NVIDIA/NVTabular/pull/925#issuecomment-876678067) of #925

Outcome:

- `_shuffle_df` is slightly more efficient
- `_add_data_scatter` now results in re-ordered data when `shuffle` is is not False

### CODE REVIEW

**Review by karlhigley — APPROVED** [review]
LGTM

(It's still weird to me that the canonical way to shuffle is to sample all the rows, but 🤷🏻 )

**Review by benfred — APPROVED** [review]

File: nvtabular/io/shuffle.py [hunk 1]
```diff
@@ -53,21 +47,11 @@ def _check_shuffle_arg(shuffle):
...
     """Shuffles a DataFrame, returning a new dataframe with randomly
     ordered rows"""
     size = size or len(df)
-    # NOTE: We can use np.arange for both gpu and cpu-backed
-    # dataframes once NEP-35 is fully accepted (`like` argument).
-    # This should be available in numpy>=1.20
     if isinstance(df, pd.DataFrame):
-        arr = np.arange(size)
-        np.random.shuffle(arr)
+        return df.sample(n=size, ignore_index=not keep_index)
```
Comment (benfred) [inline nvtabular/io/shuffle.py:55]: we're seeing a unittest failure here
```
E           TypeError: sample() got an unexpected keyword argument 'ignore_index'
```

I see this parameter on the pandas 1.3.0 docs https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.sample.html - but I don't see it on my pandas 1.2.4 install. Do we need to bump the version of pandas on our containers for this ?

Comment (rjzamora) [reply nvtabular/io/shuffle.py:55]: You're right, it was added in 1.3 - Grr
Bumping the pandas version would resolve the issue, but we could also add a `tr`y block or a version check to handle older pandas versions.  Any preference?

Comment (benfred) [reply nvtabular/io/shuffle.py:55]: cudf claims not to support pandas 1.3 so I'm not sure if we can bump the version after all =(
https://github.com/rapidsai/cudf/blob/7823a181bfbffcfb63375d8f38506d0a20696a74/conda/recipes/cudf/meta.yaml#L41 

Maybe check the pandas version to see if we can enable ```distutils.version.LooseVersion(pd.__version__) > "1.3"``` ?

**Pull request merged**